lib/Conversion/TorchToTensor/TorchToTensor.cpp

```diff
@@ -105,6 +105,43 @@ class ConvertAtenShapeToTensorPatternOp
   }
 };
 
+class ConvertAtenTensorOpPattern : public OpConversionPattern<AtenTensorOp> {
+public:
+  using OpConversionPattern<AtenTensorOp>::OpConversionPattern;
+  using OpAdaptor = typename AtenTensorOp::Adaptor;
+  LogicalResult
+  matchAndRewrite(AtenTensorOp op, OpAdaptor adaptor,
+                  ConversionPatternRewriter &rewriter) const override {
+    auto loc = op.getLoc();
+    auto list = op.getData().getDefiningOp<Torch::PrimListConstructOp>();
+    if (!list)
+      return failure();
+
+    auto typeConverter = getTypeConverter();
+    auto resultTy = cast<ShapedType>(typeConverter->convertType(op.getType()));
+    auto resultETy = resultTy.getElementType();
+
+    SmallVector<Value> values;
+    for (Value operand : list.getOperands()) {
```

Comment:
would something like this work with AtenAppendTOp or is the overall strategy just to avoid the mutating ops?

Reply:
Yes / no. We could do a canonicalizer that checks whether the input is a list constructor and replace with a new list constructor however this would not solve your issue. Mutative operations (e.g. list append) are generally hard to represent without a native types - especially for canonicalization.